p2pserver/message/utils/msg_workerpool.go

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (C) 2018 The ontology Authors
+ * This file is part of The ontology library.
+ *
+ * The ontology is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The ontology is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with The ontology.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * https://github.com/valyala/fasthttp
+ * The MIT License (MIT)
+
+ *Copyright (c) 2015-present Aliaksandr Valialkin, VertaMedia
```

Comment:
If we have to copy other's code, I suggest add them as 3rd party vendor package, rather than copy/paste.
What you think?

Reply:
The origin's implement is not common and can't directly apply to this context.